jekyll/_cci2/optimization-cookbook.md

```diff
@@ -0,0 +1,153 @@
+Caching |  This section describes how you can use different caching strategies in your implemenations to optimize your builds and workflows.
+Workflow Optimizations | This section describes how you can optimize your workflows.
+
+## Single-Threading Builds (Queuing)
+
+One of the most common tasks you may encounter when using the CircleCI platform for builds and jobs is managing multiple jobs and builds simultaneously to ensure your operations do not fail because of system timeouts. This becomes especially important when you have multiple contributors and committers working in the same environment. Because the CircleCI platform was designed to handle multiple tasks simultaneously without performance degradation or latency, concurrency may sometimes become an issue if there are a large number of jobs being queued, waiting for a previous job to be completed before the new job can be initiated, and the system timeout is set too low. In this case, one job will be completed, and other jobs may fail due to this timeout setting.
+
+To better optimize builds and jobs and prevent concurrency and subsequent jobs failing because of timeout, CircleCI has developed a single-threading (queueing) orb that specifically addresses these performance issues. By invoking this orb, you can greatly improve overall job and build performance and prevent concurrency.
+
+**Note**For more detailed information about the CircleCI Queueing orb, refer to the following CircleCI pages:
+- Queueing and Single Threading Overview - https://github.com/eddiewebb/circleci-queue
+- CircleCI Queueing Orb - https://circleci.com/orbs/registry/orb/eddiewebb/queue#quick-start
+
+### Setting Up and Configuring Your Environment to use the CircleCI Platform and CircleCI Orbs
+
+To configure your environment for the CircleCI platform and CircleCI orbs, follow the steps listed below.
+
+1) Use CircleCI `version 2.1` at the top of your `.circleci/config.yml` file.
+
+`version: 2.1`
+
+2) If you do not already have Pipelines enabled, you'll need to go to **Project Settings -> Advanced Settings** to enable pipelines.
+
+3) Add the `orbs` stanza below your version, invoking the orb:
+
+`orbs:
+  queue: eddiewebb/queue@1.1.2`
+
+4) Use `queue` elements in your existing workflows and jobs.
+
+5) Opt-in to use of third-party orbs on your organization’s **Security Settings** page.
+
+### Blocking Workflows
+
+One of the easiest ways to prevent workflow concurrency using the CircleCI Single-Threading orb is to enable "blocking" of any workflows with an earlier timestamp. By setting the `block-workflow` parameter value to `true`, all workflows will be forced to run consecutively, not concurrently, thereby limiting the number of workflows in the queue. In turn, this will also improve overall performance while making sure no workflows are discarded.
```

Comment:
What would cause a workflow to be discarded?

Reply:
Re-worded this section to make it more clear about workflow blocking and concurrency.